module/VuFind/src/VuFind/AjaxHandler/GetUserTransactions.php

```diff
@@ -57,25 +57,52 @@ public function handleRequest(Params $params)
         if (!$this->ils->checkCapability('getMyTransactions')) {
             return $this->formatResponse('', self::STATUS_HTTP_ERROR, 405);
         }
-        $items = $this->ils->getMyTransactions($patron);
+
         $counts = [
             'ok' => 0,
             'warn' => 0,
             'overdue' => 0
         ];
-        foreach ($items['records'] as $item) {
-            switch ($item['dueStatus'] ?? '') {
-            case 'due':
-                $counts['warn']++;
-                break;
-            case 'overdue':
-                $counts['overdue']++;
-                break;
-            default:
-                $counts['ok']++;
-                break;
+        $functionConfig = $this->ils->checkFunction('getMyTransactions', $patron);
+        $page = 1;
+        $pageEnd = 1;
```

Comment:
Maybe it's just me, but I find it a little confusing to set $pageEnd to 1 above the loop and then to change it inside the loop; would it be clearer to change this to a do..while loop to force the first iteration and then evaluate the condition at the bottom?